packages/dart/noports_core/lib/src/srvd/isolates/shared_single_port_isolate.dart

```diff
@@ -0,0 +1,232 @@
+import 'dart:async';
+import 'dart:io';
+import 'dart:isolate';
+import 'dart:typed_data';
+
+import 'package:at_utils/at_logger.dart';
+import 'package:noports_core/src/srv/srv_impl.dart';
+import 'package:noports_core/src/srvd/isolates/relay_worker.dart';
+import 'package:noports_core/src/srvd/isolates/types.dart';
+import 'package:noports_core/src/srvd/relay_auth_verifiers.dart';
+import 'package:noports_core/src/srvd/srvd_session_params.dart';
+import 'package:noports_core/sshnp_foundation.dart';
+import 'package:socket_connector/socket_connector.dart';
+
+class SessionInfo {
+  final SrvdSessionParams params;
+  final SocketConnector connector;
+  final Map<String, String> lookups = {};
+
+  String get atSignA => params.atSignA!;
+
+  String get atSignB => params.atSignB!;
+
+  SessionInfo(
+    this.params,
+    this.connector,
+  );
+}
+
+/// - Binds to the required port (in [run])
+/// - [startSession] handles requests from the main isolate to start sessions
+///   (sessionID, sideA atSign, sideB atSign), and is responsible for
+///   - creating the SocketConnector
+///   - Creating a SessionInfo and adding to the [sessions] Map
+///   - Removing from [sessions] when the SocketConnector is `done`
+/// - [socketHandler] handles new sockets
+///   - creates a RelayAuthVerifier for the new socket
+///   - verifies the socket's authentication
+///   - checks that the socket is for a currently live session
+///   - double-checks the socket is from one of that session's atSigns
+///   - Assigns verified sockets to that session's SocketConnector
+class SinglePortWorker extends RelayWorker {
```

Comment:
A class that does all of the actual work for a shared single port (e.g. port 443) isolate